aspnetcore/blazor/performance/webassembly-event-pipe-diagnostics.md

```diff
@@ -0,0 +1,124 @@
+---
+title: ASP.NET Core Blazor WebAssembly Event Pipe diagnostics
+author: guardrex
+description: Learn about Event Pipe diagnostics and how to get a Garbage Collector heap dump in ASP.NET Core Blazor WebAssembly apps.
+monikerRange: '>= aspnetcore-10.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/30/2025
+uid: blazor/performance/webassembly-event-pipe
+---
+# ASP.NET Core Blazor WebAssembly Event Pipe diagnostics
+
+<!-- UPDATE 10.0 - Activate ...
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+-->
+
+This article describes Event Pipe diagnostic tools, counters, and how to get a Garbage Collector heap dump in Blazor WebAssembly apps.
+
+## Prerequisite
+
+Install the [.NET WebAssembly build tools](xref:blazor/tooling/webassembly#net-webassembly-build-tools):
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+## EventPipe profiler
+
+[EventPipe](/dotnet/core/diagnostics/eventpipe) is a runtime component used to collect tracing data, similar to [ETW](/windows/win32/etw/event-tracing-portal) and [perf_events](https://wikipedia.org/wiki/Perf_%28Linux%29).
+
+* Manual testing
```

Comment:
this is notes for my team

Reply:
Not sure what you're asking me to do here ... delete? ... and if so, delete what (which lines)?

Reply:
delete whole Manual testing and Web-based testing